service/resource/vaultcrtv2/create.go

```diff
@@ -53,18 +53,12 @@ func (r *Resource) newCreateChange(ctx context.Context, obj, currentState, desir
 
 	var secretToCreate *apiv1.Secret
 	if currentSecret == nil {
-		secretToCreate = desiredSecret
-
-		err := r.ensureVaultRole(customObject)
-		if err != nil {
-			return nil, microerror.Mask(err)
-		}
-
```

Comment:
The cert resource does not need to care about the role anymore because the dedicated resource does now.